src/pages/iou/MoneyRequestWaypointPage.tsx

```diff
@@ -1,39 +1,21 @@
-import PropTypes from 'prop-types';
+import type {StackScreenProps} from '@react-navigation/stack';
 import React from 'react';
 import {withOnyx} from 'react-native-onyx';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
 import IOURequestStepWaypoint from './request/step/IOURequestStepWaypoint';
 
-const propTypes = {
-    /** The transactionID of this request */
-    transactionID: PropTypes.string,
-
-    /** Route params */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** IOU type */
-            iouType: PropTypes.string,
-
-            /** Index of the waypoint being edited */
-            waypointIndex: PropTypes.string,
-        }),
-    }),
-};
-
-const defaultProps = {
-    transactionID: '',
-    route: {
-        params: {
-            iouType: '',
-            waypointIndex: '',
-        },
-    },
+type MoneyRequestWaypointPageOnyxProps = {
+    transactionID: string | undefined;
 };
+type MoneyRequestWaypointPageProps = StackScreenProps<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT> & MoneyRequestWaypointPageOnyxProps;
 
 // This component is responsible for grabbing the transactionID from the IOU key
 // You can't use Onyx props in the withOnyx mapping, so we need to set up and access the transactionID here, and then pass it down so that WaypointEditor can subscribe to the transaction.
-function MoneyRequestWaypointPage({transactionID, route}) {
+function MoneyRequestWaypointPage({transactionID = '', route}: MoneyRequestWaypointPageProps) {
     return (
+        // @ts-expect-error TODO: Remove this once withFullTransactionOrNotFound  is migrated to TypeScript.
```

Comment:
Link the issue if it exists (same for other comments)
```suggestion
        // @ts-expect-error TODO: Remove this once withFullTransactionOrNotFound is migrated to TypeScript.
```

Reply:
there are no issues, at least in spreadsheet 
<img width="373" alt="image" src="https://github.com/Expensify/App/assets/25230417/307b401d-0294-4394-9d15-654f8b8eaa94">